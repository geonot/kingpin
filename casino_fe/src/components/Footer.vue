<template>
  <footer class="bg-gray-900 dark:bg-black text-gray-400 dark:text-gray-500 p-6 text-center text-sm mt-auto">
    <!-- Footer content -->
    <div class="container mx-auto">
        <p>&copy; {{ new Date().getFullYear() }} Kingpin Casino. All rights reserved.</p>
        <p class="mt-1">Please gamble responsibly. For help, visit BeGambleAware.org.</p>
        <!-- Optional: Links to Terms, Privacy Policy, etc. -->
        <nav class="mt-2 space-x-4">
            <router-link to="/terms" class="hover:text-gray-300">Terms</router-link>
            <router-link to="/privacy" class="hover:text-gray-300">Privacy Policy</router-link>
             <router-link to="/responsible-gaming" class="hover:text-gray-300">Responsible Gaming</router-link>
        </nav>
    </div>
  </footer>
</template>

<script setup>
// No script logic needed for simple footer
</script>

<style scoped>
/* Add specific footer styles if Tailwind isn't enough */
footer {
   /* Example: border-top: 1px solid #333; */
}
</style>


