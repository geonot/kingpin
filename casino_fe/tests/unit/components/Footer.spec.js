import { shallowMount } from '@vue/test-utils';
import Footer from '@/components/Footer.vue';

describe('Footer.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Footer, {
      global: {
        stubs: {
          'router-link': {
            template: '<a :href="to"><slot /></a>',
            props: ['to']
          }
        }
      }
    });
  });

  test('renders copyright text with current year', () => {
    const currentYear = new Date().getFullYear();
    // Using a regex to match the copyright text, including the dynamic year
    expect(wrapper.text()).toMatch(`Â© ${currentYear} Kingpin Casino. All rights reserved.`);
  });

  test('renders responsible gambling message', () => {
    expect(wrapper.text()).toContain('Please gamble responsibly. For help, visit BeGambleAware.org.');
  });

  test('renders link to Terms page', () => {
    // Find by href since we're stubbing router-link as an anchor
    const termsLink = wrapper.find('a[href="/terms"]');
    expect(termsLink.exists()).toBe(true);
  });

  test('renders link to Privacy Policy page', () => {
    const privacyLink = wrapper.find('a[href="/privacy"]');
    expect(privacyLink.exists()).toBe(true);
  });

  test('renders link to Responsible Gaming page', () => {
    const responsibleGamingLink = wrapper.find('a[href="/responsible-gaming"]');
    expect(responsibleGamingLink.exists()).toBe(true);
  });

  test('matches snapshot', () => {
    // Re-mount for snapshot if previous interactions could alter it, or use the existing wrapper
    // For a simple component like Footer, the initial mount is usually fine.
    expect(wrapper.html()).toMatchSnapshot();
  });
});
