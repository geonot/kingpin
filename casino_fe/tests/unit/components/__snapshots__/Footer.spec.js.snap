// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Footer.vue matches snapshot 1`] = `
"<footer class="bg-gray-900 dark:bg-black text-gray-400 dark:text-gray-500 p-6 text-center text-sm mt-auto">
  <!-- Footer content -->
  <div class="container mx-auto">
    <p>Â© 2025 Kingpin Casino. All rights reserved.</p>
    <p class="mt-1">Please gamble responsibly. For help, visit BeGambleAware.org.</p><!-- Optional: Links to Terms, Privacy Policy, etc. -->
    <nav class="mt-2 space-x-4"><a href="/terms" class="hover:text-gray-300">Terms</a><a href="/privacy" class="hover:text-gray-300">Privacy Policy</a><a href="/responsible-gaming" class="hover:text-gray-300">Responsible Gaming</a></nav>
  </div>
</footer>"
`;
